app/verify-email/page.tsx

```diff
@@ -1,67 +1,116 @@
+          setSubmitFailed(false)
+        }, 5000)
+      } else {
+        // To do, handle other errors apart from invalid email
+        if (response.status === 400) {
+          setSubmitFailed(true)
+        }
+      }
+    } catch (error) {
+      setSubmitFailed(true)
+    } finally {
+      setIsSubmitting(false)
+    }
+  }
 
-function verifyemail() {
   return (
-    <div className="welcome-container mx-auto flex max-w-[520px] flex-col items-center py-32">
-      <svg
-        width="56"
-        height="56"
-        viewBox="0 0 56 56"
-        fill="none"
-        xmlns="http://www.w3.org/2000/svg"
-      >
-        <circle cx="28" cy="28" r="20" fill="#7190F9" fill-opacity="0.1" />
-        <circle cx="28" cy="28" r="20" fill="#7190F9" fill-opacity="0.1" />
-        <path
-          d="M28.0003 51.3337C40.8873 51.3337 51.3337 40.8873 51.3337 28.0003C51.3337 15.1133 40.8873 4.66699 28.0003 4.66699C15.1133 4.66699 4.66699 15.1133 4.66699 28.0003C4.66699 40.8873 15.1133 51.3337 28.0003 51.3337Z"
-          fill="#4E74F8"
-          stroke="#4E74F8"
-          stroke-width="4.66667"
-          stroke-linecap="round"
-          stroke-linejoin="round"
-        />
-        <path
-          d="M16 27L23.9615 36L39 22"
-          stroke="#121317"
-          stroke-width="4.66667"
-          stroke-linecap="round"
-          stroke-linejoin="round"
-        />
-      </svg>
-
-      <div className="mt-[28px] bg-neutral-950 text-2xl"> Welcome to SigNoz </div>
-
-      <div className="text-md mt-[28px] rounded-[6px] border border-[#1D212D] bg-signoz_ink-300 p-[24px]">
-        <div>
-          {' '}
-          Thank you for signing up for SigNoz Cloud. Please check your email for the next steps.{' '}
+    <div className="welcome-container mx-auto flex min-h-[96vh] max-w-[520px] flex-col items-center justify-center py-32">
+      <Mail className="text-signoz_robin-500" size={56} />
+      <div className="mt-[28px] text-xl"> First, let's verify your email </div>
+
+      <div className="mt-[28px] rounded-[6px] border border-[#1D212D] bg-signoz_ink-300 p-[24px] text-sm">
+        Check your email for the next steps. If you don't see the email, check your spam folder or
+        try resending the verification email.
+      </div>
+
+      {submitFailed && (
+        <div className="mt-[28px] flex w-full items-center justify-center gap-2 text-center text-xs text-signoz_cherry-500">
+          <Frown size={16} /> It looks like something didn't go as planned. Please reach out to us
+          for assistance.
         </div>
+      )}
 
-        <div className="mt-[28px]">
-          If you do not find the email in your inbox within the next few minutes, please check your
-          spam folder.
+      {submitSuccess && (
+        <div className="mt-[28px] flex w-full items-center justify-center gap-2 text-center text-xs text-signoz_forest-500">
+          <CheckCircleIcon size={16} /> Verification email sent! Please check your email for the
```

Comment:
@YounixM this submit is the re-send button submit right? 

for resend use this content: 

`"Verification email resent! Please check your inbox (and spam folder) for the new verification link."`